src/dags/output.py

```diff
@@ -0,0 +1,63 @@
+import functools
+
+
+def single_output(func):
+    """Convert tuple output to single output."""
+
+    @functools.wraps(func)
+    def wrapper_single_output(*args, **kwargs):
+        raw = func(*args, **kwargs)
+        out = raw[0]
+        return out
+
+    return wrapper_single_output
+
+
+def dict_output(func=None, *, keys=None):
+    """Convert tuple output to dict output."""
+
+    def decorator_dict_output(func):
+        @functools.wraps(func)
+        def wrapper_dict_output(*args, **kwargs):
+            raw = func(*args, **kwargs)
+            out = dict(zip(keys, raw))
+            return out
+
+        return wrapper_dict_output
+
+    if callable(func):
+        return decorator_dict_output(func)
+    else:
+        return decorator_dict_output
+
+
+def list_output(func):
+    """Convert tuple output to list output."""
+
+    @functools.wraps(func)
+    def wrapper_list_output(*args, **kwargs):
+        raw = func(*args, **kwargs)
+        out = list(raw)
+        return out
+
+    return wrapper_list_output
+
+
+def aggregated_output(func=None, *, aggregator=None):
```

Comment:
I think you can even remove func here if the decorator is always called with parentheses and I like that everything would be keyword only.

Reply:
I think then I could not call it dynamically anymore, i.e

```python
g = aggregate_output(f, aggregator=lambda a, b: a + b)
```
But again, I am open to any change as long as my two usecases  work.